typing/ctype.ml

```diff
@@ -1498,14 +1499,15 @@ let instance_poly ?(keep_names=false) fixed univars sch =
 
 let instance_label fixed lbl =
   For_copy.with_scope (fun scope ->
-    let ty_res = copy scope lbl.lbl_res in
     let vars, ty_arg =
       match repr lbl.lbl_arg with
         {desc = Tpoly (ty, tl)} ->
           instance_poly' scope ~keep_names:false fixed tl ty
       | _ ->
           [], copy scope lbl.lbl_arg
     in
+    (* call [copy] after [instance_poly] to avoid introducing [Tsubst] *)
+    let ty_res = copy scope lbl.lbl_res in
```

Comment:
For the record, I don't really understand what is going on here. I suppose there is a problem when `lbl_res` and `lbl_args` share some parts of the type, but it's not clear who might introduce `Tsubst`. (I suppose `instance_poly` can introduce Tsubst nodes, and `copy` normalizes them away?)

Reply:
This is the other way round: `copy_sep` (first phase of `instance_poly`) does not introduce (and should not see) `Tsubst`'s, while `copy` does. Could add an assertion in `copy_sep`.

Reply:
We could add *many* assertions in `copy_sep`, I don't think this PR is the one where we want to start doing that.

Reply:
We are already adding assertions about `Tsubst` everywhere, so why not in `copy_sep`.
I do not pretend that this alone will make `copy_sep` self-explanatory :-)